skbio/stats/ordination/_principal_coordinate_analysis.py

```diff
@@ -137,12 +148,27 @@ def pcoa(
         eigvals, eigvecs = eigh(matrix_data)
         long_method_name = "Principal Coordinate Analysis"
     elif method == "fsvd":
-        eigvals, eigvecs = _fsvd(matrix_data, number_of_dimensions, seed=seed)
+        # new parameter for num_dimensions = number of dimensions (accounting for
+        # non-int values)
+        num_dimensions = number_of_dimensions
+        if 0 < number_of_dimensions < 1:
+            warn(
+                "FSVD: since value for number_of_dimensions is specified as float, "
+                "PCoA for all dimensions will be computed, which may "
+                "result in long computation time if the original "
+                "distance matrix is large."
+                "Consider specifying an integer value to optimize performance.",
+                RuntimeWarning,
+            )
```

Comment:
Why not progressively compute new dimensions until the cumulative variance is met?